src/components/SavedQueriesToolbar.js

```diff
@@ -0,0 +1,556 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Grid from '@material-ui/core/Grid'
+import Typography from '@material-ui/core/Typography'
+import { Selector } from './Selector'
+import { ActionsMenu } from './ActionsMenu'
+import { ExecuteButton } from './ExecuteButton'
+import Snackbar from '@material-ui/core/Snackbar'
+import classnames from 'classnames'
+
+/**
+ * Saved queries toolbar with controls and menus
+ *
+ */
+export class SavedQueriesToolbar extends React.Component {
+  static propTypes = {
+    queries: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        name: PropTypes.string,
+        query: PropTypes.string,
+        default: PropTypes.bool,
+      })
+    ),
+    selectedQueryName: PropTypes.any,
+    query: PropTypes.string,
+    handleCreateQuery: PropTypes.func,
+    handleUpdateQuery: PropTypes.func,
+    handleSelectedAction: PropTypes.func,
+    handleSelectQuery: PropTypes.func,
+    subscription: PropTypes.any,
+    handleRunQuery: PropTypes.func,
+    handleStopQuery: PropTypes.func,
+    operations: PropTypes.any,
+    handleEditQuery: PropTypes.func,
+    showActions: PropTypes.bool,
+    isActionsMenuOpen: PropTypes.bool,
+    docExplorerOpen: PropTypes.bool,
+    isOwner: PropTypes.bool,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      open: props.isActionsMenuOpen,
+      selectedQueryName:
+        props.selectedQueryName &&
+        this.findSelectedQuery(props.queries, props.selectedQueryName)
```

Comment:
`findSelectedQuery` is probably cheap enough to run a few times. We could consider to call it only once above the state assignment and then use it down here.